### PR TITLE
Add Hydrostatics

### PR DESCRIPTION
**Pull request by PartMan7**

### CODE REVIEW

File: data/mods/gen9ssb/pokedex.ts [hunk 1]
```diff
@@ -163,6 +163,14 @@ export const Pokedex: {[k: string]: ModdedSpeciesData} = {
 		abilities: {0: "Mime knows best"},
 	},
 
+	// Hydrostatics
+	pichuspikyeared: {
+		inherit: true,
+		baseStats: {hp: 35, atk: 55, def: 40, spa: 50, spd: 50, spe: 90},
+		abilities: {0: 'Hydrostatic Positivity'},
+		evos: ['Chandelure'], // Added for Eviolite compatibility
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/pokedex.ts:171]:
```suggestion
		evos: ['Chandelure'], // Added for Eviolite compatibility
```
I'd rather you just edit Eviolite to get the same effect. See pre-IoA slowpoke-galar

Comment (PartMan7) [reply data/mods/gen9ssb/pokedex.ts:171]: Did in https://github.com/smogon/ps-ssb/pull/87/commits/d83a9d3fa13806241306dcc227e2b897de2472d0

File: data/mods/gen9ssb/conditions.ts [hunk 2]
```diff
@@ -352,6 +352,23 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+		onSwitchOut() {
+			this.add(`c:|${getName('Hydrostatics')}|Brb getting some water :d`);
+		},
+		onFaint(pokemon) {
+			this.add(`c:|${getName('Hydrostatics')}|Seems like you were more hydrated than me :c`);
+			if (pokemon.side.pokemon.some(mon => mon.name === 'PartMan')) {
+				// Custom message for PartMan
+				// Yes, this reveals that the enemy has PartMan
+				this.add(`c:|${getName('PartMan')}|Hydro here have a tiara`);
+			}
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/conditions.ts:369]:
```suggestion
			if (enemyStaff(pokemon) === 'PartMan') {
				this.add(`c:|${getName('PartMan')}|Hydro here have a tiara`);
			}
```

File: data/mods/gen9ssb/conditions.ts [hunk 3]
```diff
@@ -352,6 +352,21 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
 			this.add(`c:|${getName('hsy')}|https://www.youtube.com/watch?v=g104OJIh9hs`);
 		},
 	},
+	hydrostaticsuser: {
+		noCopy: true,
+		onStart() {
+			this.add(`c:|${getName('Hydrostatics')}|Straighten your backs and get some hydration :]`);
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/conditions.ts:358]: need a typechange display

File: data/mods/gen9ssb/moves.ts [hunk 4]
```diff
@@ -824,7 +824,60 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			if (!source.terastallized) this.add('-anim', source, 'Charge Beam', source);
+			else this.add('-anim', source, 'Water Pulse', target);
+		},
+		onEffectiveness(typeMod, target, type, move) {
+			if (move.type === 'Electric') return typeMod + this.dex.getEffectiveness('Water', type);
+			else if (type === 'Water') return 1;
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/moves.ts:853]:
```suggestion
			else if (type === 'Water' && move.type === 'Water') return 1;
```

File: data/mods/gen9ssb/pokedex.ts [hunk 5]
```diff
@@ -163,6 +163,13 @@ export const Pokedex: {[k: string]: ModdedSpeciesData} = {
 		abilities: {0: "Mime knows best"},
 	},
 
+	// Hydrostatics
+	pichuspikyeared: {
+		inherit: true,
+		baseStats: {hp: 35, atk: 55, def: 40, spa: 50, spd: 50, spe: 90},
+		abilities: {0: 'Hydrostatic Positivity'},
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/pokedex.ts:170]:
```suggestion
		abilities: {0: 'Hydrostatic Positivity'},
		types: ["Electric", "Water"],
```

File: data/mods/gen9ssb/conditions.ts [hunk 6]
```diff
@@ -352,6 +352,23 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+		},
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/conditions.ts:359]:
```suggestion
		onStart() {
			this.add(`c:|${getName('Hydrostatics')}|Straighten your backs and get some hydration :]`);
			this.add('-start', pokemon, 'typechange', 'Electric/Water', '[silent]');
		},
```

**Pull request merged**